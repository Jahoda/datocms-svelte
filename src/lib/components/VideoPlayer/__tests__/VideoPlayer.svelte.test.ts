import { describe, expect, it } from 'vitest';
import '@testing-library/jest-dom';
import { render } from '@testing-library/svelte';

import { VideoPlayer } from '../../..';

describe('VideoPlayer', () => {
	describe('when data object', () => {
		describe('is complete', () => {
			const data = {
				muxPlaybackId: 'ip028MAXF026dU900bKiyNDttjonw7A1dFY',
				title: 'Title',
				width: 1080,
				height: 1920,
				blurUpThumb: 'data:image/bmp;base64,Qk0eAAAAAAAAABoAAAAMAAAAAQABAAEAGAAAAP8A'
			};

			it('unwraps data into props ready for MUX player', () => {
				const props = { data };

				const { container } = render(VideoPlayer, { props });

				expect(container).toMatchSnapshot();
			});

			describe('and `class` is passed', () => {
				it('uses it for the <mux-player /> element', () => {
					const props = { data, class: 'main-player' };

					const { container } = render(VideoPlayer, { props });

					expect(container).toMatchSnapshot();
				});
			});

			describe('and an explicitly `undefined` style is passed', () => {
				it('avoids adding aspect ratio', () => {
					const props = { data, style: undefined };

					const { container } = render(VideoPlayer, { props });

					expect(container).toMatchSnapshot();
				});
			});

			describe('and a style string is passed', () => {
				describe("that doesn't include aspect ratio", () => {
					it('adds computed aspect ratio', () => {
						const props = { data, style: 'margin: auto;' };

						const { container } = render(VideoPlayer, { props });

						expect(container).toMatchSnapshot();
					});
				});

				describe('that defines the aspect ratio property', () => {
					describe('as a valid CSS value', () => {
						it('uses the passed value to override default aspect ratio', () => {
							const props = { data, style: 'aspect-ratio: auto' };

							const { container } = render(VideoPlayer, { props });

							expect(container).toMatchSnapshot();
						});
					});
				});
			});
		});

		describe('contains `muxPlaybackId`', () => {
			const data = {
				muxPlaybackId: 'ip028MAXF026dU900bKiyNDttjonw7A1dFY'
			};

			it('uses it for `playbackId`', () => {
				const props = { data };

				const { container } = render(VideoPlayer, { props });

				expect(container).toMatchSnapshot();
			});
		});

		describe('contains `playbackId`', () => {
			const data = {
				playbackId: 'ip028MAXF026dU900bKiyNDttjonw7A1dFY'
			};

			it('uses it for `playbackId`', () => {
				const props = { data };

				const { container } = render(VideoPlayer, { props });

				expect(container).toMatchSnapshot();
			});
		});

		describe('lacks of `width` and `height` values', () => {
			const data = {
				muxPlaybackId: 'ip028MAXF026dU900bKiyNDttjonw7A1dFY',
				title: 'Title'
			};

			it('avoids adding aspect ratio', () => {
				const props = { data };

				const { container } = render(VideoPlayer, { props });

				expect(container).toMatchSnapshot();
			});
		});

		describe('lacks of `title` value', () => {
			const data = {
				muxPlaybackId: 'ip028MAXF026dU900bKiyNDttjonw7A1dFY',
				width: 1080,
				height: 1920
			};

			it('avoids adding a title', () => {
				const props = { data };

				const { container } = render(VideoPlayer, { props });

				expect(container).toMatchSnapshot();
			});
		});
	});
});
