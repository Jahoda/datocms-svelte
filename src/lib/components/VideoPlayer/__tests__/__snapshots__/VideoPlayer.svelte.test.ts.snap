// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VideoPlayer > when data object > contains \`muxPlaybackId\` > uses it for \`playbackId\` 1`] = `
<body>
  <div>
    <mux-player
      disable-cookies="true"
      playback-id="ip028MAXF026dU900bKiyNDttjonw7A1dFY"
      preload="metadata"
      stream-type="on-demand"
    />
    <!--&lt;VideoPlayer&gt;-->
  </div>
</body>
`;

exports[`VideoPlayer > when data object > contains \`playbackId\` > uses it for \`playbackId\` 1`] = `
<body>
  <div>
    <mux-player
      disable-cookies="true"
      playback-id="ip028MAXF026dU900bKiyNDttjonw7A1dFY"
      preload="metadata"
      stream-type="on-demand"
    />
    <!--&lt;VideoPlayer&gt;-->
  </div>
</body>
`;

exports[`VideoPlayer > when data object > is complete > and \`autoplay\` is passed > uses it for the <mux-player /> element 1`] = `
<body>
  <div>
    <mux-player
      autoplay="true"
      disable-cookies="true"
      placeholder="data:image/bmp;base64,Qk0eAAAAAAAAABoAAAAMAAAAAQABAAEAGAAAAP8A"
      playback-id="ip028MAXF026dU900bKiyNDttjonw7A1dFY"
      preload="metadata"
      stream-type="on-demand"
      style="aspect-ratio: 1080 / 1920;"
      title="Title"
    />
    <!--&lt;VideoPlayer&gt;-->
  </div>
</body>
`;

exports[`VideoPlayer > when data object > is complete > and \`autoplay\` is passed > with value \`false\` > doesn't use it for the <mux-player /> element 1`] = `
<body>
  <div>
    <mux-player
      disable-cookies="true"
      placeholder="data:image/bmp;base64,Qk0eAAAAAAAAABoAAAAMAAAAAQABAAEAGAAAAP8A"
      playback-id="ip028MAXF026dU900bKiyNDttjonw7A1dFY"
      preload="metadata"
      stream-type="on-demand"
      style="aspect-ratio: 1080 / 1920;"
      title="Title"
    />
    <!--&lt;VideoPlayer&gt;-->
  </div>
</body>
`;

exports[`VideoPlayer > when data object > is complete > and \`class\` is passed > uses it for the <mux-player /> element 1`] = `
<body>
  <div>
    <mux-player
      class="main-player"
      disable-cookies="true"
      placeholder="data:image/bmp;base64,Qk0eAAAAAAAAABoAAAAMAAAAAQABAAEAGAAAAP8A"
      playback-id="ip028MAXF026dU900bKiyNDttjonw7A1dFY"
      preload="metadata"
      stream-type="on-demand"
      style="aspect-ratio: 1080 / 1920;"
      title="Title"
    />
    <!--&lt;VideoPlayer&gt;-->
  </div>
</body>
`;

exports[`VideoPlayer > when data object > is complete > and \`disableCookies\` is passed > uses it for the <mux-player /> element 1`] = `
<body>
  <div>
    <mux-player
      disable-cookies="true"
      placeholder="data:image/bmp;base64,Qk0eAAAAAAAAABoAAAAMAAAAAQABAAEAGAAAAP8A"
      playback-id="ip028MAXF026dU900bKiyNDttjonw7A1dFY"
      preload="metadata"
      stream-type="on-demand"
      style="aspect-ratio: 1080 / 1920;"
      title="Title"
    />
    <!--&lt;VideoPlayer&gt;-->
  </div>
</body>
`;

exports[`VideoPlayer > when data object > is complete > and \`disableCookies\` is passed > with value \`false\` > doesn't use it for the <mux-player /> element 1`] = `
<body>
  <div>
    <mux-player
      placeholder="data:image/bmp;base64,Qk0eAAAAAAAAABoAAAAMAAAAAQABAAEAGAAAAP8A"
      playback-id="ip028MAXF026dU900bKiyNDttjonw7A1dFY"
      preload="metadata"
      stream-type="on-demand"
      style="aspect-ratio: 1080 / 1920;"
      title="Title"
    />
    <!--&lt;VideoPlayer&gt;-->
  </div>
</body>
`;

exports[`VideoPlayer > when data object > is complete > and \`preload\` is passed > uses it for the <mux-player /> element 1`] = `
<body>
  <div>
    <mux-player
      disable-cookies="true"
      placeholder="data:image/bmp;base64,Qk0eAAAAAAAAABoAAAAMAAAAAQABAAEAGAAAAP8A"
      playback-id="ip028MAXF026dU900bKiyNDttjonw7A1dFY"
      preload="auto"
      stream-type="on-demand"
      style="aspect-ratio: 1080 / 1920;"
      title="Title"
    />
    <!--&lt;VideoPlayer&gt;-->
  </div>
</body>
`;

exports[`VideoPlayer > when data object > is complete > and \`preload\` is passed > with value \`none\` > doesn't use it for the <mux-player /> element 1`] = `
<body>
  <div>
    <mux-player
      disable-cookies="true"
      placeholder="data:image/bmp;base64,Qk0eAAAAAAAAABoAAAAMAAAAAQABAAEAGAAAAP8A"
      playback-id="ip028MAXF026dU900bKiyNDttjonw7A1dFY"
      preload="none"
      stream-type="on-demand"
      style="aspect-ratio: 1080 / 1920;"
      title="Title"
    />
    <!--&lt;VideoPlayer&gt;-->
  </div>
</body>
`;

exports[`VideoPlayer > when data object > is complete > and a style string is passed > that defines the aspect ratio property > as a valid CSS value > uses the passed value to override default aspect ratio 1`] = `
<body>
  <div>
    <mux-player
      disable-cookies="true"
      placeholder="data:image/bmp;base64,Qk0eAAAAAAAAABoAAAAMAAAAAQABAAEAGAAAAP8A"
      playback-id="ip028MAXF026dU900bKiyNDttjonw7A1dFY"
      preload="metadata"
      stream-type="on-demand"
      style="aspect-ratio: auto;"
      title="Title"
    />
    <!--&lt;VideoPlayer&gt;-->
  </div>
</body>
`;

exports[`VideoPlayer > when data object > is complete > and a style string is passed > that doesn't include aspect ratio > adds computed aspect ratio 1`] = `
<body>
  <div>
    <mux-player
      disable-cookies="true"
      placeholder="data:image/bmp;base64,Qk0eAAAAAAAAABoAAAAMAAAAAQABAAEAGAAAAP8A"
      playback-id="ip028MAXF026dU900bKiyNDttjonw7A1dFY"
      preload="metadata"
      stream-type="on-demand"
      style="aspect-ratio: 1080 / 1920; margin: auto;"
      title="Title"
    />
    <!--&lt;VideoPlayer&gt;-->
  </div>
</body>
`;

exports[`VideoPlayer > when data object > is complete > unwraps data into props ready for MUX player 1`] = `
<body>
  <div>
    <mux-player
      disable-cookies="true"
      placeholder="data:image/bmp;base64,Qk0eAAAAAAAAABoAAAAMAAAAAQABAAEAGAAAAP8A"
      playback-id="ip028MAXF026dU900bKiyNDttjonw7A1dFY"
      preload="metadata"
      stream-type="on-demand"
      style="aspect-ratio: 1080 / 1920;"
      title="Title"
    />
    <!--&lt;VideoPlayer&gt;-->
  </div>
</body>
`;

exports[`VideoPlayer > when data object > lacks of \`title\` value > avoids adding a title 1`] = `
<body>
  <div>
    <mux-player
      disable-cookies="true"
      playback-id="ip028MAXF026dU900bKiyNDttjonw7A1dFY"
      preload="metadata"
      stream-type="on-demand"
      style="aspect-ratio: 1080 / 1920;"
    />
    <!--&lt;VideoPlayer&gt;-->
  </div>
</body>
`;

exports[`VideoPlayer > when data object > lacks of \`width\` and \`height\` values > avoids adding aspect ratio 1`] = `
<body>
  <div>
    <mux-player
      disable-cookies="true"
      playback-id="ip028MAXF026dU900bKiyNDttjonw7A1dFY"
      preload="metadata"
      stream-type="on-demand"
      title="Title"
    />
    <!--&lt;VideoPlayer&gt;-->
  </div>
</body>
`;
